rye/src/templates/script/default/__main__.py

```diff
@@ -0,0 +1,4 @@
+import {{ name_safe }}
```

Comment:
I know people love `__main__` but do we think it's a good idea? At least personally I really try to avoid it. Not sure how others feel about it.

Reply:
I looked into common Python packages like ensurepip, pip, etc. They all provide `__main__.py` so that you can do `python -m pip` for example. I think it is good to provide a way so people can run `python -m my-package` from a rye project. That's why i included it as a thin wrapper around __init__